src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,36 @@
+import Popup from "../components/Popup";
+
+class PopupWithConfirm extends Popup {
+  constructor({ popupSelector,}) {
+    super({ popupSelector});
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonContent = this._submitButton.textContent;
+  }
+
+
+        setSubmitAction(action) {
+            this._handleSubmitCallback = action;
+          }
+        
+          setEventListeners() {
+            this._popupElement.addEventListener("submit", (evt) => {
+              evt.preventDefault();
+              this._handleSubmitCallback();
+            });
+        
+            super.setEventListeners();
```

Comment:
You need to **format** the markup (and code) to show **indentation** and to make the code more readable. You can do that with `Prettier` or a combination: press `SHIFT + ALT + F`


```js

class PopupWithConfirm extends Popup {
  constructor({ popupSelector }) {
    super({ popupSelector });
    this._submitButton = this._popupElement.querySelector(".modal__button");
    this._submitButtonContent = this._submitButton.textContent;
  }

  setSubmitAction(action) {
    this._handleSubmitCallback = action;
  }

  setEventListeners() {
    this._popupElement.addEventListener("submit", (evt) => {
      evt.preventDefault();
      this._handleSubmitCallback();
    });

    super.setEventListeners();
  }

```